client/cli/src/runner.rs

```diff
@@ -229,10 +229,9 @@ impl<C: SubstrateCli> Runner<C> {
 	) -> Result<()> {
 		self.print_node_infos();
 		let mut task_manager = initialise(self.config)?;
-		self.tokio_runtime.block_on(main(task_manager.future().fuse()))
-			.map_err(|e| e.to_string())?;
+		let res = self.tokio_runtime.block_on(main(task_manager.future().fuse()));
 		self.tokio_runtime.block_on(task_manager.clean_shutdown());
```

Comment:
This is a recurring question, but *why* do we even need a clean shut down? 
It feels to me that someone one day has randomly put a line that waits for the shutdown, and nobody has ever dared remove it.

Reply:
TL;DR If we want to support any async runtime, we need to handle tasks cancelling.

All the details are here: https://github.com/paritytech/substrate/issues/6642

There is also a link to the issue on async-std on the matter.

Reply:
If the question is more about why not just let the tasks be... If they are not cancelled, their objects are not properly dropped (Drop is not called).

Reply:
The main issue (that came from parity-ethereum) was rocksdb corruption on improper shutdown. Don't remember if it has happened in substrate.

Reply:
That rings a bell. Maybe @arkpar would have more info on that?